src/features/session_trace/aggregate/index.js

```diff
@@ -45,54 +50,57 @@ export class Aggregate extends FeatureBase {
     this.harvestTimeSeconds = getConfigurationValue(agentIdentifier, 'session_trace.harvestTimeSeconds') || 10
     this.maxNodesPerHarvest = getConfigurationValue(agentIdentifier, 'session_trace.maxNodesPerHarvest') || 1000
     this.laststart = 0
-
+    this.mode = MODE.OFF
     const handlerCache = new HandlerCache()
-
-    registerHandler('feat-stn', () => {
-      if (typeof PerformanceNavigationTiming !== 'undefined') {
-        this.storeTiming(window.performance.getEntriesByType('navigation')[0])
-      } else {
-        this.storeTiming(window.performance.timing)
-      }
-
-      const scheduler = new HarvestScheduler('resources', {
-        onFinished: onHarvestFinished.bind(this),
-        retryDelay: this.harvestTimeSeconds
-      }, this)
-      scheduler.harvest.on('resources', prepareHarvest.bind(this))
-      scheduler.runHarvest({ needResponse: true }) // sends first stn harvest immediately
-      handlerCache.decide(true)
-
-      function onHarvestFinished (result) {
-        if (result.sent && result.responseText && !this.ptid) { // continue interval harvest only if ptid was returned by server on the first
-          getRuntime(this.agentIdentifier).ptid = this.ptid = result.responseText
-          scheduler.startTimer(this.harvestTimeSeconds)
-        }
-
-        if (result.sent && result.retry && this.sentTrace) { // merge previous trace back into buffer to retry for next harvest
-          Object.entries(this.sentTrace).forEach(([name, listOfSTNodes]) => {
-            if (this.nodeCount >= this.maxNodesPerHarvest) return
-
-            this.nodeCount += listOfSTNodes.length
-            this.trace[name] = this.trace[name] ? listOfSTNodes.concat(this.trace[name]) : listOfSTNodes
-          })
-          this.sentTrace = null
+    const sessionEntity = this.agentRuntime.session
+
+    /* --- The following section deals with user sessions concept & contains non-trivial control flow. --- */
+    if (!argsObj?.sessionTrackingOn || !sessionEntity) {
+      // Since session manager isn't around, do the old Trace behavior of waiting for RUM response to decide feature activation.
+      this.isStandalone = true
+      registerHandler('rumresp-stn', (on) => {
+        if (on === true) this.startTracing(handlerCache)
+        else handlerCache.decide(false)
+      }, this.featureName, this.ee)
+    } else {
+      const doStuffByMode = (sessionEntity) => { // take care that the state obj reference is always subject to change but session mgr keeps one instance
+        switch (sessionEntity.state.sessionTraceMode) {
+          case MODE.ERROR:
+            registerHandler('errorAgg', () => sessionEntity.state.sessionTraceMode = MODE.FULL, this.featureName, this.ee) // switch to full capture when any error is encountered
+            subscribeToVisibilityChange(visState => { // on user return, check if mode transitioned error -> full since pause (e.g., on another page), and if so, also switch here
```

Comment:
Maybe we should use the storage API to handle this so we don't miss data from waiting for vis change

Reply:
Leave as vis change for now